frozendict/tests.py

```diff
@@ -0,0 +1,131 @@
+from unittest import TestCase
+
+from frozendict import frozendict, FrozenOrderedDict
+
+
+class BaseTestCase(TestCase):
+    def assertMappingEqual(self, m1, m2):
+        # `assertDictEqual` will make a failing type check when comparing a `dict`
+        # with a `frozendict`. Instead, we'll compare the item sets:
+        self.assertSetEqual(set(m1.items()), set(m2.items()))
+
+    def assertRaisesAssignmentError(self, *args, **kwargs):
+        self.assertRaisesRegexp(
+            TypeError,
+            r'object does not support item assignment',
+            *args, **kwargs
+        )
+
+
+def setitem(dct, key, value):
+    dct[key] = value
+
+
+class TestFrozenDict(BaseTestCase):
+    def test_assignment_fails(self):
+        self.assertRaisesAssignmentError(setitem, frozendict(), 'key', 'value')
+
+    def test_init(self):
+        frozendict()
+
+    def test_init_with_keys(self):
+        fd = frozendict(a=0, b=1)
+        self.assertMappingEqual(fd, dict(a=0, b=1))
```

Comment:
I'd also be tempted to add something like:
```
fd1 = frozendict(a=0, b=1)
fd2 = frozendict(b=1, a=0)
self.assertMappingEqual(fd1, fd2)
```
(and then perhaps the opposite for `FrozenOrderedDict`)